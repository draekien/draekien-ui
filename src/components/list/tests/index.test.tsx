/** @jsx jsx */
import { jsx } from 'theme-ui';
import { render } from '@testing-library/react';
import List, { ListItemProps } from '..';
import Button from '../../button';

describe('List component', () => {
  const items = [
    { value: 1, text: 'one' },
    { value: 2, text: 'two' },
    { value: 3, text: 'three' },
  ];

  const transformFn = (item: ListItemProps) => (
    <Button fullWidth>{item.text}</Button>
  );

  const itemsWithTransform = [
    { value: 1, text: 'one', transform: transformFn },
    { value: 2, text: 'two', transform: transformFn },
    { value: 3, text: 'three', transform: transformFn },
  ];

  test('should match snapshot', () => {
    const { container } = render(<List items={items} />);

    expect(container).toMatchSnapshot();
  });

  test('should match snapshot with custom props', () => {
    const { container } = render(
      <List
        label="test"
        variant="ol"
        focusedIndex={0}
        items={itemsWithTransform}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
