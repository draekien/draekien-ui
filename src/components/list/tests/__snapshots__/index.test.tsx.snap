// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should match snapshot 1`] = `
<div>
  <div
    class="css-zkr0k6-List"
  >
    <ul
      class="css-1roqklx-List"
    >
      <li
        class="css-50zunu"
      >
        <div
          class="css-eydsp4-Text"
        >
          one
        </div>
      </li>
      <li
        class="css-50zunu"
      >
        <div
          class="css-eydsp4-Text"
        >
          two
        </div>
      </li>
      <li
        class="css-50zunu"
      >
        <div
          class="css-eydsp4-Text"
        >
          three
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`List component should match snapshot with custom props 1`] = `
<div>
  <div
    class="css-zkr0k6-List"
  >
    <div
      class="css-3vy6se-Text"
    >
      test
    </div>
    <ol
      class="css-1roqklx-List"
    >
      <li
        class="css-1ub0xpm"
      >
        <button
          class="css-c3dhtb-Button"
          type="button"
        >
          <span>
            one
          </span>
        </button>
      </li>
      <li
        class="css-50zunu"
      >
        <button
          class="css-c3dhtb-Button"
          type="button"
        >
          <span>
            two
          </span>
        </button>
      </li>
      <li
        class="css-50zunu"
      >
        <button
          class="css-c3dhtb-Button"
          type="button"
        >
          <span>
            three
          </span>
        </button>
      </li>
    </ol>
  </div>
</div>
`;
